### PR TITLE
Added connection helper and handle truncate on SQLite properly

### PR DESCRIPTION
**Pull request by tboerger**
Without the proper SQLite handling it prints errors to the log of active record.

### CODE REVIEW

**Comment by tboerger** [comment]
Travis fails just because of errors while installing rack-cache, so it's not related to this PR.

**Comment by dnrce** [comment]
Can you post an example error? Glancing at the code it looks like this would already be handled by the `rescue`.

File: lib/yaml_db/serialization_helper.rb [hunk 1]
```diff
@@ -57,16 +57,21 @@ def reenable_logger
...
           load_documents(io, truncate)
         end
       end
 
       def self.truncate_table(table)
         begin
-          ActiveRecord::Base.connection.execute("TRUNCATE #{Utils.quote_table(table)}")
+          if connection.adapter_name == "SQLite"
+            connection.execute("DELETE FROM #{Utils.quote_table(table)}")
+            connection.execute("VACUUM")
```
Comment (dnrce) [inline lib/yaml_db/serialization_helper.rb]: Given the performance penalty, is `VACUUM` appropriate to do here? Aren't we always refilling the database (not necessarily with as much data, I'll admit)?

Plus, this kind of optimization generally seems a bit out of scope for what yaml_db is doing.

**Comment by tboerger** [comment]
I don't have this error message anymore because I'm already using this pull request. But you can see always that this query failed.

File: lib/yaml_db/serialization_helper.rb (hunk 1 above)
Comment (tboerger) [inline lib/yaml_db/serialization_helper.rb]: DELETE and VACUUM are the suggested queries to have the same behavior as TRUNCATE on other databases. I can't see any performance issues. We are already using this fix within a backup and restore functionality.

**Comment by tboerger** [comment]
So no progress here? If that won't get accepted I will simply close this PR and keep the patch on top of our SUSE packages.

**Comment by tboerger** [comment]
I'm closing this PR because of lacking activity. Feel free to integrate it on your own. I like a clean pull request listing on my side, nearly 3 months for a PR is too long for my taste ;)

**Pull request closed without merging**